### PR TITLE
URL to initiate SAML authentication flow

### PR DESCRIPTION
**Pull request by devkiran**
## What does this PR do?

This page is used to initiate the SAML authentication flow by redirecting to the SAML provider. Accessible only on self-hosted Cal.com instances.

Visit `/auth/sso/direct` to initiate the SAML authentication flow.

**Environment**: Staging(main branch) / Production

## Type of change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by leog** [comment]
@devkiran is there an easy way to test this locally?

**Comment by devkiran** [comment]
@leog Please follow the steps below to test this locally:

- Enable the SAML SSO feature as described in the [README](https://github.com/calcom/cal.com/tree/main/packages/features/ee#setting-up-saml-login)
- For the IdP XML Metadata you can use the [metadata.xml](https://mocksaml.com/api/saml/metadata)
- Add a new SAML connection at _Settings_ > _Single Sign-On_ > _SSO with SAML_ and paste the metadata content.
- Logout from the app
- Visit `/auth/sso/direct` to initiate the SAML authentication flow.

**Comment by leog** [comment]
Thanks @devkiran, it works as expected.
Could you please clarify the purpose of this feature? Is it to improve SSO access from outside?

**Review by leog — APPROVED** [review]

File: apps/web/pages/auth/sso/direct.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { signIn } from "next-auth/react";
+import { useRouter } from "next/router";
+
+import { samlProductID, samlTenantID } from "@calcom/features/ee/sso/lib/saml";
+import { HOSTED_CAL_FEATURES } from "@calcom/lib/constants";
+
+import { inferSSRProps } from "@lib/types/inferSSRProps";
```
Comment (emrysal) [inline apps/web/pages/auth/sso/direct.tsx:7]:
```suggestion
import type { inferSSRProps } from "@lib/types/inferSSRProps";
```

**Pull request merged**